lib/api/src/sys/module.rs

```diff
@@ -149,7 +149,12 @@ impl Module {
     /// the WebAssembly text format (if the "wat" feature is enabled for
     /// this crate).
     pub fn from_binary(store: &impl AsStoreRef, binary: &[u8]) -> Result<Self, CompileError> {
-        Self::validate(store, binary)?;
+        if let Err(err) = Self::validate(store, binary) {
+            if let Ok(module) = unsafe { Self::deserialize(store, binary) } {
+                return Ok(module);
+            }
```

Comment:
From binary should only take `.wasm` / `wat` formats (not the serialized artifact of Wasmer/Engine).

Reply:
Leftover from a refactor, replaced it